test/fixtures/detect-existing-browser-api-object/content.js

```diff
@@ -33,3 +33,22 @@ test("browser api object in background page", async (t) => {
     t.ok(!reply.windowBrowserIsUnchanged, "window.browser API object should have been defined by the polyfill");
   }
 });
+
+test("error types", async (t) => {
+  if (navigator.userAgent.includes("Firefox/")) {
+    try {
+      await browser.storage.sync.set({a: 'a'.repeat(10000000)});
```

Comment:
Based on the error message, it seems that you can remove the keys themselves. The API call is rejected without even trying to store the data.